plugins/grid-bookmark/src/GridBookmarkWidget/components/AssemblySelector.tsx

```diff
@@ -1,71 +1,70 @@
 import React from 'react'
 import { observer } from 'mobx-react'
 
-import { Checkbox, MenuItem, SelectChangeEvent, TextField } from '@mui/material'
+import {
+  Select,
+  Button,
+  MenuItem,
+  FormControl,
+  InputLabel,
+  Checkbox,
+  ListItemText,
+  OutlinedInput,
+  ListItemIcon,
+} from '@mui/material'
 
 // locals
 import { GridBookmarkModel } from '../model'
-import { makeStyles } from 'tss-react/mui'
-import { getSession } from '@jbrowse/core/util'
 
-const useStyles = makeStyles()({
-  textfield: {
-    padding: 5,
-  },
-})
-
-const GridBookmarkAssemblySelector = observer(function AssemblySelector({
+const GridBookmarkAssemblySelector = observer(function ({
   model,
 }: {
   model: GridBookmarkModel
 }) {
-  const { assemblyManager } = getSession(model)
-  const { classes } = useStyles()
-  const { validAssemblies, selectedAssemblies, setSelectedAssemblies } = model
-  const allSelected =
-    validAssemblies.size > 0 &&
-    selectedAssemblies.length === validAssemblies.size
-
-  const handleChange = (event: SelectChangeEvent<string[]>) => {
-    const value = event.target.value
-    if (value.at(-1) === 'All') {
-      setSelectedAssemblies(
-        selectedAssemblies.length === validAssemblies.size
-          ? []
-          : [...validAssemblies],
-      )
-      return
-    }
-    setSelectedAssemblies([...value])
-  }
-
+  const { validAssemblies, selectedAssemblies } = model
+  const noAssemblies = validAssemblies.size === 0 ? true : false
+  const label = 'Select assemblies'
   return (
-    <TextField
-      select
-      variant="outlined"
-      fullWidth
-      className={classes.textfield}
-      SelectProps={{
-        multiple: true,
-        value: selectedAssemblies,
-        // @ts-ignore
-        onChange: handleChange,
-        // @ts-ignore
-        renderValue: selected => selected.join(', '),
-      }}
-      label="Select assembly"
-    >
-      <MenuItem value="All">
-        <Checkbox checked={allSelected} />
-        All
-      </MenuItem>
-      {[...validAssemblies].map(name => (
-        <MenuItem key={name} value={name}>
-          <Checkbox checked={selectedAssemblies.includes(name)} />
-          {assemblyManager.get(name)?.displayName || name}
+    <FormControl disabled={noAssemblies} fullWidth>
+      <InputLabel id="select-assemblies-label">{label}</InputLabel>
+      <Select
+        labelId="select-assemblies-label"
+        id="select-assemblies"
+        multiple
+        value={selectedAssemblies}
+        onChange={event => model.setSelectedAssemblies([...event.target.value])}
+        input={<OutlinedInput label={label} />}
+        renderValue={selected => selected.join(', ')}
+      >
+        <MenuItem value="all">
+          <Checkbox
+            checked={[...validAssemblies].every(e =>
+              selectedAssemblies.includes(e),
+            )}
```

Comment:
the "Select all" Checkbox now has it's own onChange event handler, this make's it so that the main Select component onChange handler has easier behavior and avoids specialized behavior on the name "All" which could be a name of an assembly...who knows. does require stopPropagation (which i generally try to avoid, but it is just out of the feeling that stopping event propagation can be a big hammer to wield), but it does help here